app/services/change_schedule.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+class ChangeSchedule
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+  include Participants::ProfileAttributes
+
+  attribute :cpd_lead_provider
+  attribute :participant_id
+  attribute :reason
+  attribute :course_identifier
+  attribute :schedule_identifier
+  attribute :cohort
+
+  alias_attribute :user_profile, :participant_profile
+  delegate :participant_profile_state, to: :participant_profile, allow_nil: true
+  delegate :lead_provider, to: :cpd_lead_provider, allow_nil: true
+
+  validates :schedule, presence: { message: I18n.t(:invalid_schedule) }
+  validate :not_already_withdrawn
+  validate :validate_new_schedule_valid_with_existing_declarations
+  validate :validate_provider
+  validate :validate_permitted_schedule_for_course
+  validate :validate_cannot_change_cohort
+  validate :schedule_valid_with_pending_declarations
+
+  def call
+    return if invalid?
+
+    ActiveRecord::Base.transaction do
+      ParticipantProfileSchedule.create!(participant_profile:, schedule:)
+      participant_profile.update_schedule!(schedule)
+
+      if relevant_induction_record
+        Induction::ChangeInductionRecord.call(
+          induction_record: relevant_induction_record,
+          changes: {
+            schedule:,
+          },
+        )
+      end
+    end
+
+    relevant_induction_record_for_profile(participant_profile)
```

Comment:
let's also have a chat about this, I introduced a method particularly for serialisation, as I think this will return `nil` now for NPQ participants when changing schedule